code/modules/crew_objectives/engineering_objectives.dm

```diff
@@ -32,3 +32,41 @@
 			if(istype(dumbbird.ears, /obj/item/radio/headset))
 				return TRUE
 	return ..()
+
+/datum/objective/crew/supermatter_survive
+	explanation_text = "Prevent the primary Supermatter engine from fully delaminating. Additional Supermatter shards ordered from cargo do not count for this"
+	jobs = "chiefengineer,stationengineer"
+
+/datum/objective/crew/supermatter_survive/check_completion()
+	if(GLOB.main_supermatter_engine)  // handy it qdel's itself
+		return TRUE
+	return ..()
+
+
+/datum/objective/crew/apc
+	explanation_text = "Make sure the station has above (something broke message us)kW in the powernet."
+	jobs = "chiefengineer,stationengineer,atmospherictechnician"
+
+/datum/objective/crew/apc/New()
+	. = ..()
+	target_amount = rand(100,500)  // no i dont know how much a normal amount is :>
+	explanation_text = "Make sure the station has above [target_amount]kW in the powernet."
+
+/datum/objective/crew/apcc/check_completion()
+	var/total_power = 0
+	var/list/powermonitors = list()
+
+	for(var/obj/machinery/computer/monitor/pMon in GLOB.machines)
+		if(pMon.stat & (NOPOWER | BROKEN)) //check to make sure the computer is functional
+			continue
+		if(pMon.is_secret_monitor) //make sure it isn't a secret one (ie located on a ruin), allowing people to metagame that the location exists
+			continue
+		powermonitors += pMon
+
+	for(var/obj/machinery/computer/monitor/pMon in powermonitors)
```

Comment:
We don't need two loops what is done here can go in the previous one, I also don't think you are doing the power checking in a proper way, make sure you get the station powernet and only add it once to the count.

Reply:
ok whats the proper way to get the current powernet? i know very little about pnet code

Reply:
Make sure you get a power monitor connected to the station and only add up the available power once.

Reply:
*how do i do that*

Reply:
Just check which powermonitor is connected to the powernet with the most power and use only that one I suppose

Reply:
yea what commands do i use

Reply:
Just see which of the monitors you are looping through has the most power and check for that one